File: backend/src/main/java/turip/creator/controller/dto/response/CreatorResponse.java
package turip.creator.controller.dto.response;

import turip.creator.domain.Creator;

public record CreatorResponse(
        Long id,
        String channelName,
        String profileImage
) {

    public static CreatorResponse from(Creator creator) {
        return new CreatorResponse(
                creator.getId(),
                creator.getChannelName(),
                creator.getChannelName()
        );
    }
}

File: android/app/src/main/java/com/on/turip/data/creator/dto/CreatorResponse.kt
package com.on.turip.data.creator.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class CreatorResponse(
    @SerialName("channelName")
    val channelName: String,
    @SerialName("id")
    val id: Long,
    @SerialName("profileImage")
    val profileImage: String,
)
